tuf/tuf_test.go

```diff
@@ -988,25 +868,20 @@ func TestGetDelegationRolesInvalidPaths(t *testing.T) {
 
 	testKey1, err := ed25519.Create("targets/test", data.ED25519Key)
 	assert.NoError(t, err)
-	role, err := data.NewRole("targets/test", 1, []string{testKey1.ID()}, []string{"path", "anotherpath"})
-	assert.NoError(t, err)
 
-	err = repo.UpdateDelegations(role, data.KeyList{testKey1})
+	err = repo.UpdateDelegations("targets/test", changelist.TufDelegation{AddKeys: []data.PublicKey{testKey1}, AddPaths: []string{"path", "anotherpath"}, NewThreshold: 1})
 	assert.NoError(t, err)
 
 	testKey2, err := ed25519.Create("targets/test/b", data.ED25519Key)
 	assert.NoError(t, err)
-	// Now we add a delegation with a path that is not prefixed by its parent delegation
-	role, err = data.NewRole("targets/test/b", 1, []string{testKey2.ID()}, []string{"invalidpath"})
-	assert.NoError(t, err)
-
-	err = repo.UpdateDelegations(role, data.KeyList{testKey2})
+	// Now we add a delegation with a path that is not prefixed by its parent delegation, the invalid path won't be added
+	err = repo.UpdateDelegations("targets/test/b", changelist.TufDelegation{AddKeys: []data.PublicKey{testKey2}, AddPaths: []string{"invalidpath"}, NewThreshold: 1})
 	assert.NoError(t, err)
 
-	// Getting this delegation does not actually restrict paths, unless we use the RestrictChild method
+	// Updating the delegation will restrict paths
 	delgRole, err := repo.GetDelegationRole("targets/test/b")
 	assert.NoError(t, err)
-	assert.Contains(t, delgRole.Paths, "invalidpath")
+	assert.NotContains(t, delgRole.Paths, "invalidpath")
```

Comment:
Updating delegations on a repo validates paths now, so the `invalidpath` won't be added to the role in the repo